crates/matrix-sdk-store-encryption/src/lib.rs

```diff
@@ -774,8 +774,7 @@ impl StoreCipher {
     }
 }
 
-#[derive(Zeroize)]
-#[zeroize(drop)]
+#[derive(ZeroizeOnDrop)]
```

Comment:
I'm not quite certain, but I think we want `#[derive(Zeroize, ZeroizeOnDrop)]`: https://docs.rs/zeroize/latest/zeroize/index.html#custom-derive-support.

Not exactly sure what only deriving `ZeroizeOnDrop` does, I would need to let Rust expand the macro: https://github.com/RustCrypto/utils/blob/3867a73955c0c35bd3bd066d086c07c5e43a78c8/zeroize_derive/src/lib.rs#L99-L120.

Reply:
This is `Zeroize`:

```
// ========================================
// Recursive expansion of the Zeroize macro
// ========================================

impl ::zeroize::Zeroize for MacKey {
    fn zeroize(&mut self) {
        match self {
            #[allow(unused_variables)]
            MacKey(__zeroize_field_0) => __zeroize_field_0.zeroize(),
            _ => {}
        }
    }
}
```

vs `ZeroizeOnDrop`:

```
// ==============================================
// Recursive expansion of the ZeroizeOnDrop macro
// ==============================================

impl Drop for MacKey {
    fn drop(&mut self) {
        use ::zeroize::__internal::AssertZeroize;
        use ::zeroize::__internal::AssertZeroizeOnDrop;
        match self {
            #[allow(unused_variables)]
            MacKey(__zeroize_field_0) => __zeroize_field_0.zeroize_or_on_drop(),
            _ => {}
        }
    }
}
#[doc(hidden)]
impl ::zeroize::ZeroizeOnDrop for MacKey {}
```

so I think this is doing the right thing?

Reply:
Alright, though I can't think of a reason why we wouldn't want `Zeroize` and the `zeroize()` method to be available.